src/codegen/doc/mod.rs

```diff
@@ -503,8 +514,39 @@ where
             )?;
         }
 
-        for parameter in fn_.parameters() {
-            if parameter.instance_parameter || parameter.name.is_empty() {
+        // A list of parameters positions to filter out
+        let mut indices_to_ignore: BTreeSet<_> = fn_
+            .parameters()
+            .iter()
+            .filter_map(|param| param.array_length)
+            .collect();
+        if let Some(indice) = fn_.ret().array_length {
+            indices_to_ignore.insert(indice);
+        }
+
+        // The original list of parameters without the ones that specifics an array length
+        let no_array_length_params: Vec<_> = fn_
+            .parameters()
+            .iter()
+            .enumerate()
+            .filter_map(|(indice, param)| {
+                (!indices_to_ignore.contains(&(indice as u32))).then(|| param)
+            })
+            .filter(|param| !param.instance_parameter)
+            .collect();
+
+        let in_parameters = no_array_length_params.iter().filter(|param| {
+            let ignore = IGNORED_C_FN_PARAMS.contains(&param.name.as_str())
+                || param.direction == ParameterDirection::Out
+                // special case error pointer as it's transformed to a Result
+                || (param.name == "error" && param.c_type == "GError**")
+                // special case `data` as it could be something else other than `user_data`
```

Comment:
```suggestion
                // special case `data` with explicit `gpointer` type as it could be something else (unlike `user_data`)
```

I think is more fitting for what you're trying to convey here?

Reply:
yes, what i wrote was more of frenglish :P